node/config/load.go

```diff
@@ -46,7 +72,51 @@ func FromReader(reader io.Reader, def interface{}) (interface{}, error) {
 	return cfg, nil
 }
 
-func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool) ([]byte, error) {
+// function from raw config string to loadability
+type loadabilityCheck struct {
+	configOk       func(string) error
+	configNotFound func() error
+}
+
+func DefaultFullNodeLoadabilityCheck() loadabilityCheck {
+	configOk := func(cfgRaw string) error {
+		poisonStr := "#EnableSplitstore =" // splitstore must be set explicitly
+		if strings.Contains(cfgRaw, poisonStr) {
+			return xerrors.Errorf("Config contains disallowed string %s, refusing to load. Please explicitly set EnableSplitstore. Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default", poisonStr)
+		}
+		return nil
+	}
+	configNotFound := func() error {
```

Comment:
Having a specific comment about the splitstore here isn't going to age well.

Reply:
And shouldn't the default config handle this for us?

Reply:
If the user reads this error message, they're going to come to slack and say "wtf, help!".

Reply:
> If the user reads this error message, they're going to come to slack and say "wtf, help!".

I kind of wrote this expecting that to be fine.  I thought of this as going out of our way to really make sure people don't lose data with unexpected discard mode even if they *delete their config file* which I didn't think was common at all.

That said I can add in instructions to regenerate a default config file and warn that splitstore will be set in that config.

Reply:
We definitely don't want to be editing/changing defaults. But yeah, a quick "run this command to get unstuck" would be good.

Reply:
Ok added `lotus config default` instruction.  Reiterating that we don't want the default config handling this edge case for us while transition from splitstore and that its ok to reference splitstore specifically in the error.  Its not going to age well but were also going to remove this error and that message in a few months when everyone has updated their node and dealt with this.